examples/react-hooks-example/src/post.md

```diff
@@ -0,0 +1,221 @@
+
+`last5$` creates a new observable that returns the last 5 trades streamed from `EthPurchase$` using the [`latest` operator](https://github.com/embarklabs/subspace/blob/353fee7d23f12ffa3a8f91fe095500335016586e/packages/core/src/operators.js#L84) that is exposed by Subspace.
+
+### Setting up subscriptions
+
+Now, let's fire up the subscriptions.
+
+First, let's start streaming `EthPurchase` events.  This is a very simple example and just starts up the subscription and prints each transaction to the developer console.  An important thing to remember is to pass the observable's `unsubscribe` function in the return value of the effect.  This is an important feature of Effect hooks.  The hook runs at each component re-render or whenever an element in the dependency array changes and so its important to clean up the subscription before the re-render to avoid memory leaks.  React will call the function specified in the return before re-running the hook during re-render.
+
+```js
+    useEffect(() => {
+        if ((txnObserver === undefined) || (typeof latestBlock != "number")) {
+            return;
+        } else {
+            txnObserver.subscribe((trade) => {
+                console.log(trade);
+            });
+        }
+        return txnObserver.unsubscribe;
+    }, [txnObserver, latestBlock]);
+```
+
+Next, the magic.  Our `last5Observer` observable is the one that produces the data we actually use in the dapp's UI.
+
+```js
+    useEffect(() => {
+        if (last5Observer === undefined) {
+            return;
+        } else {
+            last5Observer.subscribe((fiveTrades) => {
+                const prices = fiveTrades.map(trade => {
+                    const txnDetails = new TradeDetails(trade.tokens_sold, trade.eth_bought);
+                    return {'block': trade.blockNumber, 'rate': txnDetails.exchangeRate}
+                });
+                setLast5(prices);
+            });
+        }
+        return last5Observer.unsubscribe;
+    }, [last5Observer]);
+```
+
+The magic here is that under the hood,  Subspace stores all the events that are streamed from `EthPurchase$` in a local database and then `last5Observer` uses the `latest` operator to grab the "latest" 5 events and store them in the `last5` state variable maintained by the `App` component's state hook we set up earlier.  Inside the subscribe function, we're transforming the data in the `EthPurchase` events to provide the block number of each transaction and its exchange rate.  Any experts reading this will know that you can use RxJS operators like `pipe` and `map` to transform this data within the initial observable declaration but the above approachs makes it clear what we're doing. 
+
+### Displaying the data
+
+Now, let's do something with this data.  Since this is just a demo, we have only 2 visualizations of the data sourced from Subspace but this should give you an idea how powerful Subspace is.
```

Comment:
```suggestion
Now, let's do something with this data.  Since this is just a demo, we have only two visualizations of the data sourced from Subspace but this should give you an idea how powerful Subspace is.
```